src/utils/prepareResults.ts

```diff
@@ -18,6 +18,8 @@ export const defaultComputeResultFingerprint: ComputeFingerprintFunc = (rule, ha
 
   if (rule.source) id += rule.source;
 
+  if (rule.message) id += rule.message;
```

Comment:
@marbemac I''ve had to patch the fingerprinting function which was too agressive fro this new test cases. Does that fit you?

Reply:
Interesting - which test case(s) fail if this is not added? The fingerprint already takes into account the rule code (`typed-enum`) and the error range/path - is there a situation in which we would want two `typed-enum` errors reported at the same path? Or maybe I'm not understanding the underlying issue.

Reply:
@marbemac The rule relies on a function which analyzes properties exposing both a `type` and an `enum`. The function return an array of messages. Each message points at an invalid enum value. 

```OpenAPI 3.x detected

{document}
 7:12  error  typed-enum  Enum value "Tap, tap, tap..." do not respect the specified type "integer".
 7:12  error  typed-enum  Enum value "Is thing thing on?" do not respect the specified type "integer".
 7:12  error  typed-enum  Enum value "You certainly wonder" do not respect the specified type "integer".
 7:12  error  typed-enum  Enum value "Why I gathered you all today..." do not respect the specified type "integer".

✖ 4 problems (4 errors, 0 warnings, 0 infos, 0 hints)
```

Without that change, only one error message would be returned above, as the returned results only differ by the message.

Reply:
Ah interesting - I guess the new question is do we want to do it this way? Versus a single error that indicates there are enum values that don't match.

```
7:12  error  typed-enum  Enum values "Tap, tap, tap...", "Is thing thing on?",  "You certainly wonder", and "Why I gathered you all today..." do not respect the specified types "integer".
```

Don't have a super strong opinion on this, but it would mean the assumption of one error per file location + error code is still a rule that holds true. Other potential benefit is that it results in less reporting noise since the occurrences would be grouped together instead of reported as distinct errors.

Will leave it to you and @P0lip / @philsturgeon to decide :).

Reply:
@philsturgeon So, what's your vote on this? Revert to a reporting threshold (to prevent long list from generating a too long long error message)? Or stick with current implementation which may have the unexpected (good?) side effect of fixing #920 as well?

I understand @marbemac 's point in that concatenating would result in a terser reporting, although I wouldn't tag it as noise, as each error message would point at a different real issue.

Reply:
@nulltoken @marbemac I think it depends on how you view the rule conceptually. Are we reporting that the enum is invalid because it has some bad items listed in it? Or are we reporting that the array item is invalid? More often than not I see people using the muli line YAML synax:

```
1: type: boolean
2: enum: 
3: - true
4: - nonsense
```

I would expect to see error message for line 4 not line 2, because line 4 is the problem not line 2. 

if there are 3 problematic lines in the enum, I would expect 3 errors, because each problem is another thing to fix.

Reply:
If we go w @philsturgeon's suggestion, then `range` would be different per reported error, and original fingerprint strategy should work as is.

However, it's been a while since I've been in this are of Spectral codebase, so I'm not sure how difficult it would be to report path/range at an array index resolution (`['myProp', 'enum', 1]`) rather than at a array property resolution (`['myProp', 'enum']`).

Reply:
@philsturgeon @marbemac  Indeed, it makes much more sense to point at the value rather at the enum level. On it!

Reply:
@philsturgeon @marbemac 
- The rule now reports the ranges of the offending values
- The change to the fingerprinting function has been reverted

Reply:
Very nice! Moving forward, IF we're ever in another situation where we're considering expanding the fingerprint function, it might indicate a separate issue :). I'm hopeful we can hold true to the "rule code + range/path" combo is unique in a result set.